shuffle/cli/src/node.rs

```diff
@@ -1,20 +1,36 @@
 // Copyright (c) The Diem Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::shared::read_config;
+use crate::shared::get_shuffle_dir;
 use anyhow::Result;
 use diem_config::config::NodeConfig;
-use std::path::Path;
+use diem_genesis_tool::validator_builder::ValidatorConfig;
+use diem_types::on_chain_config::VMPublishingOption;
+use std::{fs, path::Path};
 
-pub fn handle(project_path: &Path) -> Result<()> {
+pub fn handle() -> Result<()> {
     // TODO: Generate prefunded accounts
-    let node_config = NodeConfig::load(project_path.join("nodeconfig/0/node.yaml").as_path())?;
-    let config = read_config(project_path)?;
-    println!(
-        "running shuffle node configured for {} in {}",
-        &config.blockchain,
-        project_path.display()
-    );
+    let node_config_path = &get_shuffle_dir();
+    if !Path::new(node_config_path).is_dir() {
+        println!("Creating node config in {}", node_config_path.display());
+        fs::create_dir_all(node_config_path)?;
+        generate_validator_config(node_config_path)?;
+    } else {
+        println!("Accessing node config in {}", node_config_path.display());
+    }
+    let node_config = NodeConfig::load(node_config_path.join("nodeconfig/0/node.yaml").as_path())?;
+    println!("running shuffle node in {}", node_config_path.display());
     diem_node::start(&node_config, None);
     Ok(())
 }
+
+fn generate_validator_config(node_config_path: &Path) -> Result<ValidatorConfig> {
+    let publishing_option = VMPublishingOption::open();
+    // TODO (dimroc): place genesis module deployment/generate validator config
+    // in shuffle node command
+    shuffle_custom_node::generate_validator_config(
+        node_config_path.join("nodeconfig").as_path(),
+        diem_framework_releases::current_module_blobs().to_vec(),
+        publishing_option,
+    )
+}
```

Comment:
Can we have at least one unit test here, to cover the creation of the validator config? Another useful one would be the skipping of config creation. Refactor for testability by placing code in testable functions.

Reply:
yes!